lib/views/survey/survey_field.dart

```diff
@@ -0,0 +1,375 @@
+// ignore_for_file: avoid_renaming_method_parameters, type_annotate_public_apis
+
+import 'package:flutter/material.dart';
+import 'package:provider/provider.dart';
+import 'package:thc/models/theme.dart';
+import 'package:thc/views/survey/survey_questions.dart';
+import 'package:thc/views/survey/survey_screen.dart';
+
+/// {@template views.survey.SurveyField}
+/// A widget that displays a survey question.
+/// {@endtemplate}
+class SurveyField extends StatelessWidget {
+  /// {@macro views.survey.SurveyField}
+  const SurveyField(this.record, this.update, {super.key});
+
+  /// {@macro views.survey.SurveyRecord}
+  final SurveyRecord record;
+
+  /// {@template ValueChanged}
+  /// For a very long time, I had no idea what [ValueChanged] was;
+  /// then I realized it's just a function signature.
+  ///
+  /// 2 ways to write the same thing:
+  /// ```dart
+  /// final ValueChanged<int> onChanged;
+  /// final void Function(int) onChanged;
+  /// ```
+  /// {@endtemplate}
+  final ValueChanged<dynamic> update;
+
+  @override
+  Widget build(BuildContext context) {
+    final builder = SurveyBuilder.fromRecord(record);
+    final question = _QuestionText(record.question);
+
+    Widget answer = builder.buildAnswer(context, update, record.question, record.cleanAnswer);
+    if (record.question is MultipleChoice) {
+      answer = _MultipleChoiceTheme(child: answer);
+    }
+
+    return _ErrorBox(
+      valid: record.valid || !context.watch<SurveyValidation>().state,
+      child: builder.layout(context, question, answer),
+    );
+  }
+}
+
+/// {@template views.survey.SurveyRecord}
+/// Extension types are great for when you want to make an existing type behave in a new way.
+///
+/// `SurveyRecord` takes a [Record] of question and answer data
+/// and has methods that can validate the input and output a description of the answer.
+/// {@endtemplate}
+extension type SurveyRecord.fromRecord((SurveyQuestion, dynamic) record) {
```

Comment:
When I first read about extension types on [dart.dev](https://dart.dev/language/extension-types), it seemed like they were kinda useless.

But since then, I've seen the light 🙏✨